engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -2868,6 +2876,82 @@ public boolean restartNetwork(final Long networkId, final Account callerAccount,
         }
     }
 
+    @Override
+    public VirtualRouter findExpendableRouterForRollingRestart(final List<? extends VirtualRouter> routers) {
+        VirtualRouter expendableRouter = null;
+        if (routers != null && routers.size() > 1) {
```

Comment:
You invert the logic here. `if (router == null || router.size <=1)`, then return null;
This would allow you to remove one nesting level.